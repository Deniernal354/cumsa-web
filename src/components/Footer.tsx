import SocialLinks from "./SocialLinks";

export default function Footer() {
  return (
    <footer className="mt-12 border-t border-black/10">
      <div className="bg-white">
        <div className="container-base py-5 flex flex-col md:flex-row md:items-center md:justify-between gap-4">
          <div className="flex items-center gap-3">
            <span
              aria-hidden
              className="logo-mask-blue rounded-sm"
              style={{ width: 28, height: 28 }}
            />
            <div className="text-sm text-gray-700">
              <div>
                <a className="hover:underline" href="mailto:info@cumsa.ca">
                  info@cumsa.ca
                </a>
              </div>
              <div>Â© {new Date().getFullYear()} CUMSA</div>
            </div>
          </div>
          <SocialLinks className="text-[var(--navy)]" />
        </div>
      </div>
    </footer>
  );
}
