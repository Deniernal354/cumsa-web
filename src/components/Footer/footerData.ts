export type SecretLink = { label: string; href: string; external?: boolean };
export type SecretGroup = { label: string; links: SecretLink[] };
export type SecretSection = {
  title: string;
  links?: SecretLink[];
  groups?: SecretGroup[];
};

export const SECRET_SECTIONS: SecretSection[] = [
  {
    title: "",
    links: [{ label: "Release Notes", href: "/extras/release-notes" }],
  },
  {
    title: "Pages Drafts",
    groups: [
      {
        label: "About Us",
        links: [{ label: "Gallery", href: "/about-us/gallery" }],
      },
      {
        label: "Services",
        links: [
          { label: "New Muslims", href: "/services/new-muslims" },
          { label: "Roommate Services", href: "/services/roommate" },
          { label: "Mental Health", href: "/services/mental-health" },
        ],
      },
    ],
  },
];
