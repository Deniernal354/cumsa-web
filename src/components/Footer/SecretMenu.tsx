"use client";
import { useEffect, useRef } from "react";
import { SECRET_SECTIONS } from "./footerData";

interface SecretMenuProps {
  anchorRef: React.RefObject<HTMLElement | HTMLAnchorElement | null>;
  onClose: () => void;
}

export default function SecretMenu({ anchorRef, onClose }: SecretMenuProps) {
  const menuRef = useRef<HTMLDivElement | null>(null);

  useEffect(() => {
    const handleClick = (e: MouseEvent) => {
      if (!menuRef.current) return;
      if (menuRef.current.contains(e.target as Node)) return;
      if (anchorRef.current && anchorRef.current.contains(e.target as Node)) return;
      onClose();
    };
    window.addEventListener("mousedown", handleClick);
    return () => window.removeEventListener("mousedown", handleClick);
  }, [onClose, anchorRef]);

  return (
    <div
      ref={menuRef}
      role="menu"
      aria-label="Secret footer menu"
      className="absolute z-50 bottom-10 left-0 w-72 max-h-[70vh] overflow-y-auto rounded-lg border border-black/10 bg-white shadow-lg p-3 flex flex-col gap-4 text-sm animate-fadeIn"
    >
      {SECRET_SECTIONS.map((section) => (
        <div key={section.title} className="space-y-2">
          <div className="font-semibold text-[var(--navy)] text-xs uppercase tracking-wide opacity-80">
            {section.title}
          </div>
          <div className="space-y-1">
            {section.links?.map((l) => (
              <MenuLink key={l.href} {...l} />
            ))}
            {section.groups?.map((g) => (
              <DetailsGroup key={g.label} label={g.label} links={g.links} />
            ))}
          </div>
        </div>
      ))}
      <button
        type="button"
        onClick={onClose}
        className="mt-1 self-end text-xs px-2 py-1 rounded bg-gray-100 hover:bg-gray-200"
      >
        Close
      </button>
    </div>
  );
}

function MenuLink({ label, href, external }: { label: string; href: string; external?: boolean }) {
  if (external) {
    return (
      <a
        href={href}
        target="_blank"
        rel="noopener noreferrer"
        className="block rounded px-2 py-1 hover:bg-[var(--blue)]/10 hover:text-[var(--blue)]"
      >
        {label}
      </a>
    );
  }
  return (
    <a
      href={href}
      className="block rounded px-2 py-1 hover:bg-[var(--blue)]/10 hover:text-[var(--blue)]"
    >
      {label}
    </a>
  );
}

function DetailsGroup({
  label,
  links,
}: {
  label: string;
  links: { label: string; href: string; external?: boolean }[];
}) {
  return (
    <details className="group border rounded">
      <summary className="list-none cursor-pointer select-none px-2 py-1 flex items-center justify-between">
        <span>{label}</span>
        <span className="transition group-open:rotate-90 text-xs">â–¶</span>
      </summary>
      <div className="pb-2">
        {links.map((l) => (
          <MenuLink key={l.href} {...l} />
        ))}
      </div>
    </details>
  );
}
