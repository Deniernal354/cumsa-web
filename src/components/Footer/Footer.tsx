"use client";
import { useEffect, useRef, useState } from "react";
import SocialLinks from "@/components/SocialLinks/SocialLinks";
import SecretMenu from "./SecretMenu";

export default function Footer() {
  const [open, setOpen] = useState(false);
  const triggerRef = useRef<HTMLAnchorElement | null>(null);

  useEffect(() => {
    if (!open) return;
    const onKey = (e: KeyboardEvent) => {
      if (e.key === "Escape") setOpen(false);
    };
    window.addEventListener("keydown", onKey);
    return () => window.removeEventListener("keydown", onKey);
  }, [open]);

  return (
    <footer className="relative mt-12 border-t border-black/10">
      <div className="bg-white">
        <div className="container-base py-5 flex flex-col md:flex-row md:items-center md:justify-between gap-4">
          <div className="relative flex items-center gap-3">
            <span
              aria-hidden
              className="logo-mask-blue rounded-sm"
              style={{ width: 28, height: 28 }}
            />
            <div className="text-sm text-gray-700">
              <a
                ref={triggerRef}
                href="/release-notes"
                onClick={(e) => {
                  e.preventDefault();
                  setOpen((o) => !o);
                }}
                className="hover:underline focus:underline outline-none cursor-pointer"
                aria-haspopup="menu"
                aria-expanded={open}
              >
                Â© {new Date().getFullYear()} CUMSA
              </a>
            </div>
            {open && <SecretMenu anchorRef={triggerRef} onClose={() => setOpen(false)} />}
          </div>
          <SocialLinks className="text-[var(--navy)]" />
        </div>
      </div>
    </footer>
  );
}
