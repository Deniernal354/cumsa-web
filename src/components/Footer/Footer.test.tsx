import { render } from "@testing-library/react";
import Footer from "./Footer";

test("Footer snapshot", () => {
  const { asFragment } = render(<Footer />);
  expect(asFragment()).toMatchSnapshot();
});
