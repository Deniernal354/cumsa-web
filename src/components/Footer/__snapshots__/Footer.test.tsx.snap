// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer snapshot 1`] = `
<DocumentFragment>
  <footer
    class="relative mt-12 border-t border-black/10"
  >
    <div
      class="bg-white"
    >
      <div
        class="container-base py-5 flex flex-col md:flex-row md:items-center md:justify-between gap-4"
      >
        <div
          class="relative flex items-center gap-3"
        >
          <span
            aria-hidden="true"
            class="logo-mask-blue rounded-sm"
            style="width: 28px; height: 28px;"
          />
          <div
            class="text-sm text-gray-700"
          >
            <a
              aria-expanded="false"
              aria-haspopup="menu"
              class="hover:underline focus:underline outline-none cursor-pointer"
              href="/release-notes"
            >
              Â© 2025 CUMSA
            </a>
          </div>
        </div>
        <div
          class="text-[var(--navy)]"
          data-testid="stub"
        />
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
